pkg/controller/compute/droplet.go

```diff
@@ -144,14 +143,30 @@ func (c *dropletExternal) Create(ctx context.Context, mg resource.Managed) (mana
 
 	cr.Status.SetConditions(xpv1.Creating())
 
-	name := meta.GetExternalName(cr)
+	name, id := do.GetNameAndID(cr, cr.Name)
 
-	create := &godo.DropletCreateRequest{}
-	docompute.GenerateDroplet(name, cr.Spec.ForProvider, create)
+	var droplet *godo.Droplet
+	var errString string
+	var err error
+
+	if id != -1 {
```

Comment:
We typically would handle the case where the resource already exists in Observe as it is called first and is already attempting to Get the Droplet. Is there anything about a Droplet that makes this more suited to be handled in Create?

Reply:
Would create just bail if external name is already set then? Only thing I can think of is that we'd only fill in the `ForProvider` fields in create and not observe normally.

Reply:
Updated to us observe and late init instead!

Reply:
Nice! 🙌🏻